core/node/rules/can_add_event.go

```diff
@@ -404,9 +405,13 @@ func (params *aeParams) canAddMemberPayload(payload *StreamEvent_MemberPayload)
 			params:      params,
 			fulfillment: content.KeyFulfillment,
 		}
-		return aeBuilder().
+		ruleBuilderAE := aeBuilder().
 			checkOneOf(params.creatorIsMember).
 			check(ru.validKeyFulfillment)
+		if _, err := params.streamView.(events.ChannelStreamView).GetChannelInception(); err == nil {
+			ruleBuilderAE = ruleBuilderAE.requireChainAuth(params.channelMessageReadEntitlements)
+		}
```

Comment:
can you use the same check that we use above? 
```
		} else if shared.ValidChannelStreamId(ru.params.streamView.StreamId()) {
```

Reply:
Just so we're clear... this already requires the user to be a member of the channel
- how is this check different from the membership requirement?
- how is it not redundant? especially if we get the scrubber working

Reply:
9:52 AM
I thought that last week we agreed that the stream node itself should not only check that the user is a member, but also thta they have channel read permissions, before accepting a key fulfillment request.





But now that I think about it maybe I misunderstood our conversation, and that even if a malicious node inserts a bad join stream for a channel, a non-malicious node will not accept this event into the minipool, is that right?
so if the channel stream join event went through this node's validation then we wouldn't really need to check read permissions again.

Trying to understand your perspective here. Is that where you're coming from?

Reply:
no i'm remembering that we might want to proactively kick a user out of a stream if they fail permissions checks.
does this set us up for that?

Reply:
Oh that's true... I wasn't thinking about implementing that here, and I think it might require some architectural changes. let's discuss?

Reply:
Can you use the same check we use above?

else if shared.ValidChannelStreamId(ru.params.streamView.StreamId())